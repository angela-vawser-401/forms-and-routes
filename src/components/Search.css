.Search {
  display: flex;
  flex-direction: column;
  align-items: center;

  h2 {
    text-transform: uppercase;
    font-family: sans-serif;
    font-weight: 700;
    font-size: 1.35em;
    margin-top: 2em;
    margin-bottom: .75em;
  }

  input {
    line-height: 20px;
    width: 300px;
    margin-bottom: 10px;
  }
  
  button {
    text-transform: uppercase;
    padding: 5px;
    background-color: black;
    font-weight: 700;
    color: white;
    font-size: 16px;
    display: block;
    width: 150px;
  }
}
