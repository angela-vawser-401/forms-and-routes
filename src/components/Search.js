import React from 'react';
import PropTypes from 'prop-types';
import styles from './Search.css';

const Search = ({ search, handleSubmit, handleChange }) => {

  return (
    <form className={styles.Search} onSubmit={handleSubmit}>
      <h2>Search Characters</h2>
      <input type="text" name="search" value={search} onChange={handleChange}></input>
      <button>Search</button>
    </form>
  );
};

Search.propTypes = {
  search: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default Search;
