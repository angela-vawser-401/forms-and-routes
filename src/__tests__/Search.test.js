import React from 'react';
import { shallow } from 'enzyme';
import Search from '../components/Search';

describe('Search component', () => {
  it('renders Search', () => {
    const wrapper = shallow(<Search
      search="hello"
      handleChange={() => { }}
      handleSubmit={() => { }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
