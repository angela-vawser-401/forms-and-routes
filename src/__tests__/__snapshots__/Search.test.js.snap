// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
<form
  className="Search"
  onSubmit={[Function]}
>
  <h2>
    Search Characters
  </h2>
  <input
    name="search"
    onChange={[Function]}
    type="text"
    value="hello"
  />
  <button>
    Search
  </button>
</form>
`;
